components/brave_shields/browser/ad_block_filters_provider_manager.cc

```diff
@@ -46,7 +46,7 @@ void AdBlockFiltersProviderManager::AddProvider(
   auto it =
       std::find(filters_providers_.begin(), filters_providers_.end(), provider);
```

Comment:
`std::find` will do a linear search, the point of using a set is you can use either `base::Contains` or the methods of flat_set itself